workers/unity/Packages/io.improbable.gdk.debug/WorkerInspector/EntityList.cs

```diff
@@ -0,0 +1,268 @@
+
+        private readonly Entities entities = new Entities();
+
+        private int lastViewVersion;
+
+        public EntityList()
+        {
+            const string uxmlPath = "Packages/io.improbable.gdk.debug/WorkerInspector/Templates/EntityList.uxml";
+            var template = AssetDatabase.LoadAssetAtPath<VisualTreeAsset>(uxmlPath);
+            template.CloneTree(this);
+
+            listView = this.Q<ListView>();
+            listView.itemHeight = 24;
+            listView.makeItem = () => new Label();
+            listView.bindItem = BindItem;
+            listView.onSelectionChanged += OnSelectionChanged;
+            listView.itemsSource = entities.Data;
+
+            var searchField = this.Q<ToolbarSearchField>();
+            searchField.RegisterCallback<ChangeEvent<string>>(OnSearchFieldChanged);
+        }
+
+        public void Update()
+        {
+            if (entitySystem == null)
+            {
+                listView.itemsSource = new List<EntityId>();
+                return;
+            }
+
+            if (entitySystem.ViewVersion != lastViewVersion)
+            {
+                entities.RefreshData(world);
+                listView.itemsSource = entities.Data;
+                lastViewVersion = entitySystem.ViewVersion;
+            }
+        }
+
+        public void SetWorld(World world)
+        {
+            this.world = world;
+            entitySystem = world?.GetExistingSystem<EntitySystem>();
+            lastViewVersion = 0;
+        }
+
+        private void BindItem(VisualElement element, int index)
+        {
+            var entity = entities.Data[index];
+
+            if (!(element is Label label))
+            {
+                return;
+            }
+
+            label.text = entity.ToString();
+        }
+
+        private void OnSelectionChanged(List<object> selections)
+        {
+            // TODO: Error handling
+            // We only allow you to select one at a time
+            var entityData = (EntityData) selections[0];
+            OnEntitySelected?.Invoke(entityData.EntityId);
+        }
+
+        private void OnSearchFieldChanged(ChangeEvent<string> changeEvent)
+        {
+            var searchValue = changeEvent.newValue.Trim();
+            SearchParameters searchParameters;
+            if (long.TryParse(searchValue, out var value))
+            {
+                searchParameters = new SearchParameters { EntityId = new EntityId(value) };
+            }
+            else
+            {
+                searchParameters = new SearchParameters { SearchFragment = searchValue.ToLower() };
+            }
+
+            entities.SearchParameters = searchParameters;
+            entities.Calculate();
+            listView.itemsSource = entities.Data;
+        }
+
+        private struct SearchParameters
+        {
+            public EntityId? EntityId;
+            public string SearchFragment;
+        }
+
+        private class Entities
```

Comment:
Could this class name be a little clearer?

Reply:
Renamed to `EntityListData` and moved to its own file